### PR TITLE
Add integration tests

### LINKED ISSUE #121: Add  integration tests
docker compose + fake api

### PR DESCRIPTION
**Pull request by dron22**
This PR adds integration tests, which executes the basic wakatime-cli commands with a real binary against a test server mock api. The following commands are covered:

- send heartbeats
- today summary
- today goal
- useragent
- version

Closes #121

### CODE REVIEW

**Review by alanhamlett — APPROVED** [review]

**Comment by dron22** [comment]
@gandarez FYI, I separated unit test run from integration test run and would go forward and merge this now.

File: docker/docker-compose.windows.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: "3.8"
+
+services:
+  wakatime-cli:
+    container_name: wakatime-cli
+    build:
+      context: .
+      dockerfile: ./docker/windows.Dockerfile
```
[the remark this answers is not in the document]
Comment (dron22) [reply docker/docker-compose.windows.yml:8]: I named them according to the naming pattern suggested by  in official docker gh repos. Check for reference: https://github.com/docker/docker.github.io/issues/9245 + https://github.com/docker/docker.github.io/pull/10562

File: .github/workflows/on_push.yml [hunk 2]
```diff
@@ -11,48 +11,78 @@ env:
...
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: "Checkout"
+        uses: actions/checkout@v2
+      -
+        name: "Docker compose run"
+        run: >-
+          docker-compose
+          -f ./docker/docker-compose.linux.yml
```
[the remark this answers is not in the document]
Comment (dron22) [reply .github/workflows/on_push.yml:50]: I prefer to have them in a distinct folder out of the following reasons:
- The Dockerfiles and docker-compose files are only used for running integration tests.
- IMO, having them in repo root, would suggest, they can be used to run the service locally.
- Also we have multiple files for the different operating systems, which would bloat repo root.

[the remark this answers is not in the document]
Comment (gandarez) [reply .github/workflows/on_push.yml:50]: Can we have only one `docker compose` file? By having only one wil simplify things and you can run it by calling `docker-compose run <service-name>`.  For example:

```yml
version: "3.8"

services:
  wakatime-cli-linux:
    container_name: wakatime-cli-linux
    build:
      context: .
      dockerfile: ./docker/linux.Dockerfile

wakatime-cli-windows:
    container_name: wakatime-cli-windows
    build:
      context: .
      dockerfile: ./docker/windows.Dockerfile
```

- Linux -> `docker-compose run wakatime-cli-linux`
- Windows `docker-compose run wakatime-cli-windows`

File: docker/docker-compose.windows.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gandarez) [reply docker/docker-compose.windows.yml:8]: Never seen Dockerfile named like that, it's totally new to me.

**Comment by dron22** [comment]
@gandarez I found a simpler way, which is more fitting to our use case:
- I moved binary building to the github workflow
- The fact that docker is bound to the linux kernel limits us in testing on different operating systems. Using standard github action runners, we can effectively test on all major operating systems: ubuntu, windows, macos
- This way, we also do not need any docker/docker-compose related files :slightly_smiling_face:

**Comment by dron22** [comment]
@gandarez The macOS problem, which you mentioned is also fixed and all macOS tests pass: https://github.com/wakatime/wakatime-cli/actions/runs/837106636

File: .github/workflows/on_push_protected_branches.yml [hunk 3]
```diff
@@ -25,8 +25,8 @@ jobs:
         name: "Pull dependencies"
         run: go mod vendor
       - 
-        name: "Unit tests"
-        run: make test
+        name: "Tests"
+        run: make test-integration
```
Comment (gandarez) [inline .github/workflows/on_push_protected_branches.yml:29]: Actually it should be `make test` right?

File: .github/workflows/on_push_protected_branches.yml [hunk 4]
```diff
@@ -51,8 +51,8 @@ jobs:
...
```
Comment (gandarez) [inline .github/workflows/on_push_protected_branches.yml:55]: The same here.

**Review by gandarez — CHANGES_REQUESTED** [review]
Don't we need also to add integration tests to `on_push_protected_branches` workflow?

File: .github/workflows/on_push_protected_branches.yml (hunk 3 above)
Comment (dron22) [reply .github/workflows/on_push_protected_branches.yml]: Good catch on integration tests PR :+1: I completely forgot to adjust the protected branches workflow.

Now looking at it, I wonder if it makes sense to copy all these unit tests/integration tests and have the same identical code in 2 workflows. And actually there is an option to run jobs conditionally. Here we could use this to have everything in one workflow and run the release job only for develop and `master branches.

Check latest commit, which implements this approach.

**Comment by dron22** [comment]
@gandarez ping

**Review by gandarez — APPROVED** [review]
LGTM

**Pull request merged**